frontend/simple/model/contracts/events.js

```diff
@@ -41,10 +42,17 @@ export class GroupContract extends Events.HashableGroup {
           let memberCount = Object.keys(state.profiles).length
           let threshold = Math.ceil(state.proposals[data.proposalHash].threshold * memberCount)
           if (state.proposals[data.proposalHash].against.length > memberCount - threshold) {
+            // TODO: flag instead of delete to make proposal history easier?
             Vue.delete(state.proposals, data.proposalHash)
           }
         }
       },
+      HashableGroupCloseProposal (state, {data}) {
+        if (state.proposals[data.proposalHash]) {
+          // TODO: flag instead of delete to make proposal history easier?
+          Vue.delete(state.proposals, data.proposalHash)
```

Comment:
Yes, let me summarize the sorts of data that we have that I've documented so far:

1. Ephemeral data - data that appears in the `data ()` of a vue component but doesn't (necessarily) get saved to the store or the database
2. Persistent data - data that appears in the database and the store

We should now discuss two types of persistent data:

1. Type 1: Persistent data that is "in-memory" and synced between the vuex `$store` and the database
2. Type 2: Persistent data that is retrieved on-demand from the database. If it ever appears in the vuex `$store`, it is only temporarily, and likely only partially.

**The history of proposals is Type 2 persistent data.**

In other words, there is no need to keep it in RAM all of the time. The only time we need it is to display the historical proposals, and even then, only show about 20 results at a time.

Another example of Type 2 data that we already have is the history of the entire contract chain for all of the contracts.

I will create an issue for this and link it to this comment.

Reply:
https://github.com/okTurtles/group-income-simple/issues/426